asynch/proto/streams/buffered.py

```diff
@@ -142,7 +142,11 @@ async def read_varint(self):
             packets.append(packet)
             if packet < 0x80:
                 break
-        return leb128.u.decode(packets)
+
+        if packets:
+            return leb128.u.decode(packets)
+        else:
+            return None
```

Comment:
When the server gracefully terminates the connection, we read nothing from the stream. However then `leb128.u.decode(packets)` would decode an empty `bytearray()` into `0`, which is later interpreted as the Hello packet, causing a `UnexpectedPacketFromServerError`.

Reply:
Just in case, graceful termination is the only way to get empty bytearray here? meaning, are we sure we don't break some other scenario, etc?

Reply:
Honestly, I'm not sure. However, decoding an empty bytearray into 0 is most definitely always false. So this might break something, but it will only break things that have already been broken in the first place.